test/CMakeLists.txt

```diff
@@ -7,43 +7,62 @@ target_include_directories(${PROJECT_NAME}_TestPlugins1
   PUBLIC "../include" ${console_bridge_INCLUDE_DIRS} ${Poco_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME}_TestPlugins1 ${PROJECT_NAME})
 class_loader_hide_library_symbols(${PROJECT_NAME}_TestPlugins1)
-
+if(WIN32)
+target_compile_definitions(${PROJECT_NAME}_TestPlugins1 PRIVATE "CLASS_LOADER_BUILDING_DLL")
+endif()
 add_library(${PROJECT_NAME}_TestPlugins2 EXCLUDE_FROM_ALL SHARED plugins2.cpp)
 target_include_directories(${PROJECT_NAME}_TestPlugins2
   PUBLIC "../include" ${console_bridge_INCLUDE_DIRS} ${Poco_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME}_TestPlugins2 ${PROJECT_NAME})
+if(WIN32)
+target_compile_definitions(${PROJECT_NAME}_TestPlugins2 PRIVATE "CLASS_LOADER_BUILDING_DLL")
+endif()
 class_loader_hide_library_symbols(${PROJECT_NAME}_TestPlugins2)
 
-ament_add_gtest(${PROJECT_NAME}_utest utest.cpp)
+set(append_library_dirs "${CMAKE_CURRENT_BINARY_DIR}")
+
+if(WIN32)
+  set(append_library_dirs "${append_library_dirs}/../$<CONFIG>;${append_library_dirs}/$<CONFIG>")
+endif()
+
+ament_add_gtest(${PROJECT_NAME}_utest utest.cpp
+  APPEND_LIBRARY_DIRS "${append_library_dirs}"
+)
+
 if(TARGET ${PROJECT_NAME}_utest)
   target_include_directories(${PROJECT_NAME}_utest
     PUBLIC "../include" ${console_bridge_INCLUDE_DIRS} ${Poco_INCLUDE_DIRS})
-  target_link_libraries(${PROJECT_NAME}_utest ${PROJECT_NAME} ${Poco_LIBRARIES})
+  target_link_libraries(${PROJECT_NAME}_utest
+    ${PROJECT_NAME}
+    ${PROJECT_NAME}_TestPlugins1
+    ${PROJECT_NAME}_TestPlugins2
+    ${Poco_LIBRARIES}
+  )
   if(NOT WIN32)
     target_link_libraries(${PROJECT_NAME}_utest pthread)
   endif()
-  add_dependencies(${PROJECT_NAME}_utest ${PROJECT_NAME}_TestPlugins1 ${PROJECT_NAME}_TestPlugins2)
+  add_dependencies(${PROJECT_NAME}_utest
+    ${PROJECT_NAME}_TestPlugins1
+    ${PROJECT_NAME}_TestPlugins2)
```

Comment:
Since the target already links against these libraries the dependency declaration is redundant.

Same below.

Reply:
indeed the target should not link these libraries, fixed in 08d4345335c1b4b64e4761a490e3136ba87d36c4